### PR TITLE
Add additional .flake8 ignores and fix S607/B607

### PR DESCRIPTION
**Pull request by konstruktoid**
This PR:
- adds assert and subprocess ignores to .flake8 since we will be using `assert` and `subprocess`
- fixes flake8/bandit S607/B607 (https://flake8.codes/S607/)

### CODE REVIEW

File: tests/test_vagrant.py [hunk 1]
```diff
@@ -641,13 +645,11 @@ def _execute_command_in_vm(v, command):
     Run command via ssh on the test vagrant box.  Returns a tuple of the
     return code and output of the command.
     """
-    vagrant_exe = vagrant.get_vagrant_executable()
-
-    if not vagrant_exe:
+    if not VAGRANT_EXE:
         raise RuntimeError(vagrant.VAGRANT_NOT_FOUND_WARNING)
```
Comment (apatard) [inline tests/test_vagrant.py:649]: Do we still need this test ? after all, if vagrant is not there, earlier tests like test_boxesvm would already have failed

File: .flake8 [hunk 2]
```diff
@@ -51,6 +51,11 @@ extend-ignore =
   D205,
   D400,
   D401,
+  DUO116, # use of "shell=True" is insecure in "subprocess" module
```
Comment (ssbarnea) [inline .flake8:54]: Where are these coming from? Clearly not from from default set of plugins.

Comment (apatard) [reply .flake8:54]: don't know. About the shell=True thing, I've a patch somewhere for that and I plan to submit it once the CI is fine.

Comment (konstruktoid) [reply .flake8:54]: https://github.com/dlint-py/dlint/blob/master/docs/linters/DUO116.md

i'm using https://github.com/konstruktoid/action-pylint/blob/master/requirements.txt, but if @apatard got a future patch for it why not leave it in untill merged?

File: tests/test_vagrant.py (hunk 1 above)
Comment (konstruktoid) [reply tests/test_vagrant.py:649]: true, will fix

**Review by ssbarnea — APPROVED** [review]

File: .flake8 (hunk 2 above)
Comment (apatard) [reply .flake8:54]: shell removal pushed as PR112.

**Comment by konstruktoid** [comment]
removed DUO116 since it will be fixed in #112

**Pull request merged**